include/dxc/DxilContainer/DxilPipelineStateValidation.h

```diff
@@ -1078,6 +1083,9 @@ DxilPipelineStateValidation::ReadOrWrite(const void *pBits, uint32_t *pSize,
   if (mode == RWMode::CalcSize) {
     *pSize = rw.GetSize();
     m_pPSVRuntimeInfo1 = nullptr; // clear ptr to tempRuntimeInfo
+  } else if (mode == RWMode::Read) {
+    if (rw.GetSize() != rw.GetOffset())
+      return false;
```

Comment:
We don't want to fail the load when the size isn't equal to the version, otherwise, the forward compatibility will be broken.  We only want to verify this in the validator, to ensure that we only validate known versions of PSV.

Reply:
Changed to return rw.GetOffset() and do the compare in validator.

Reply:
We'll fail the load when PSVVersion > MAX_PSV_VERSION anyway.
So, fail the load when size not match should be fine as well.

Reply:
PSVVersion could send as default. So fail the load will not work :(